### PR TITLE
Holybro CM4 IP address corrections and UXRCE-DDS addition

### PR DESCRIPTION
**Pull request by farhangnaderi**
I am doing this PR to update ip addresses and UXRCE DDS for this specific board.

### CODE REVIEW

File: en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md [hunk 1]
```diff
@@ -212,6 +212,22 @@ Ethernet connectivity provides a fast, reliable, and flexible communication alte
 
 ::: info
 For more general information see: [PX4 Ethernet Setup](../advanced_config/ethernet_setup.md).
+Except we have used the below ```netplan``` config:
```
Comment (hamishwillee) [inline en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:215]: You only need one backtick inline

File: en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md [hunk 2]
```diff
@@ -212,6 +212,22 @@ Ethernet connectivity provides a fast, reliable, and flexible communication alte
...
+
+```sh
```
Comment (hamishwillee) [inline en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:217]: Just FYI (and you may have done this) I would add why you need this configuration. I presume it is because this is what comes "out of the box" and it is incompatible with PX4 out of the box. 

I won't make other comments until this is ready for review.

Comment (farhangnaderi) [reply en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:217]: The route is touched with the docs from Ethernet setup which disconnects internet access on pi. We either need to edit Ethernet Setup docs to this or just mention on this specific board. 
The best way to me is to correct ethernet setup doc.

Comment (farhangnaderi) [reply en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:217]: I added the reason for it under the note.

File: en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md [hunk 3]
```diff
@@ -205,13 +205,33 @@ On the RPi side:
...
+The flight controller module is [internally connected to RPi CM4](#rpi-cm4-fc-serial-connection) from `TELEM2` (serial).
 
 You can also set up a local Ethernet connection between them using the supplied cable.
 Ethernet connectivity provides a fast, reliable, and flexible communication alternative to using USB or other serial connections.
 
 ::: info
-For more general information see: [PX4 Ethernet Setup](../advanced_config/ethernet_setup.md).
+For general Ethernet setup information see: [PX4 Ethernet Setup](../advanced_config/ethernet_setup.md).
+
+The setup here is much the same, eccept that we have used the following `netplan` config on PX4:
```
Comment (hamishwillee) [inline en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:216]:
```suggestion
The setup here is much the same, except that we have used the following `netplan` config on PX4:
```

File: en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md (hunk 2 above)
Comment (hamishwillee) [reply en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:218]: Unfortunately, you might as well be speaking in another language. 

I think you are saying that the generic Ethernet docs cover other boards, and to make things correct for this board we need to do some things different. Right?

File: en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md [hunk 4]
```diff
@@ -205,13 +205,33 @@ On the RPi side:
...
+        - 10.41.10.1/24
+      nameservers:
+        addresses: [10.41.10.1]
+      routes:
+        - to: 10.41.10.0/24  # Local route to access devices on this subnet
+          scope: link        # Scope link to restrict it to local subnet
+```
+
+This sets `eth0` as our channel for the local Ethernet link.
+Note that we only set up this locally linked route to prevent internet access disruption over WiFi.
```
Comment (hamishwillee) [inline en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:234]:
```suggestion
This sets `eth0` as our channel for the local Ethernet link (instead of ``, which is used in [the normal setup](../advanced_config/ethernet_setup.md#ubuntu-ethernet-network-setup).
Note that we only set up this locally linked route to prevent internet access disruption over WiFi.
```

File: en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md [hunk 5]
```diff
@@ -252,130 +272,141 @@ You can also try to enable it manually:
...
 ```
 
-This means the CM4's ethernet IP is 169.254.21.183.
+This means the CM4's ethernet IP is `10.41.10.1` .
 
-#### IP Setup on FC
+#### Ping Test
 
-Now connect to the NuttX shell (using a console, or the MAVLink shell), and check the status of the link:
+First from the CM4:
```
Comment (hamishwillee) [inline en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:292]:
```suggestion
First ping PX4 from the CM4 (using the PX4's default address):
```

File: en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md [hunk 6]
```diff
@@ -252,130 +272,141 @@ You can also try to enable it manually:
...
-Now check the config again:
-
-```sh
-ifconfig
-
-eth0    Link encap:Ethernet HWaddr xx:xx:xx:xx:xx:xx at UP
-        inet addr:0.0.0.0 DRaddr:192.168.0.254 Mask:255.255.255.0
-```
+:::info
+If this step is failing it is worth to check if the [firewall](https://wiki.ubuntu.com/UncomplicatedFirewall) is active.
```
Comment (hamishwillee) [inline en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:312]:
```suggestion
If this step fails, check if a [firewall](https://wiki.ubuntu.com/UncomplicatedFirewall) is active.
```

File: en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md [hunk 7]
```diff
@@ -252,130 +272,141 @@ You can also try to enable it manually:
...
+:::
 
-However, it doesn’t have an IP yet.
-Set one that is similar to the one on the RPi CM4:
+Then from the flight controller in Nuttx Shell:
```
Comment (hamishwillee) [inline en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:315]:
```suggestion
Then ping the CM4 from the flight controlle.
Enter the following command in the Nuttx Shell:
```

File: en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md [hunk 8]
```diff
@@ -252,130 +272,142 @@ You can also try to enable it manually:
...
-56 bytes from 169.254.21.183: icmp_seq=9 time=0 ms
-10 packets transmitted, 10 received, 0% packet loss, time 10010 ms
-```
+To run a MAVSDK example, install mavsdk via pip, and try out an example from [MAVSDK-Python/examples](https://github.com/mavlink/MAVSDK-Python/tree/main/examples).
 
-#### MAVLink/MAVSDK Test
+#### XRCE-Client Ethernet Setup
 
-For this, we need to set the MAVLink instance to send traffic to the CM4's IP address:
+Now it is needed to enable MAVLink on the PX4 `Ethernet` port and enable `XRCE-DDS`.
```
Comment (hamishwillee) [inline en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:356]: What are you doing here? I THINK you are setting up MAVLink to run on the wired connection to TELEM2, and XRCE-DDS to connect to the Ethernet link you just set up right?

File: en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md (hunk 2 above)
Comment (farhangnaderi) [reply en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:218]: > I think you are saying that the generic Ethernet docs assumes that the wired Ethernet connection is on `enp2s0` but for RPi it is `eth0` - is that right?

Correct!

File: en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md [hunk 9]
```diff
@@ -205,13 +205,33 @@ On the RPi side:
...
+        - 10.41.10.1/24
+      nameservers:
+        addresses: [10.41.10.1]
+      routes:
+        - to: 10.41.10.0/24  # Local route to access devices on this subnet
+          scope: link        # Scope link to restrict it to local subnet
+```
+
+This sets `eth0` as our channel for the local Ethernet link from the RPi (instead of `enp2s0`, which is assumed in [Ethernet Setup](../advanced_config/ethernet_setup.md#ubuntu-ethernet-network-setup)).
+Note that we only set up this locally linked route to prevent internet access disruption over WiFi.
```
[the remark this answers is not in the document]
Comment (farhangnaderi) [reply en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:234]: > Is this what you meant?
> 
> What does the second sentence mean? Perhaps you are suggesting that it is some kind of backup route?

We do not want the default route to be over local ethernet network cause then it blocks internet access. So we need to keep the default route over WiFi to have internet access. This is what I came up to with testing.

File: en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md (hunk 8 above)
Comment (farhangnaderi) [reply en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:356]: > What are you doing here? I THINK you are setting up MAVLink to run on the wired connection to TELEM2, and XRCE-DDS to connect to the Ethernet link you just set up right?

I am enabling Ethernet on MAV_2 here.

File: en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (hamishwillee) [reply en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:234]: @farhangnaderi Assume I know nothing. What is the/a default route? Do you mean the default route set on RPi or something else? What makes something the default route? Is it the name `eth0` or `enp2s0`?

What I "guess" you are saying is that we don't want to send Internet traffic over our local ethernet network because we're already using it for XRCE-DDS. If we were to send our internet packets the whole thing would fall over. But what are these internet packets you are sending? Connection to QGC over Wifi?

Anyway, so I think you're setting up a local link to send our XRCE-DDS packets over ETH, so we can keep our Wifi link open to QGC. Is that right? Presumably by setting up `eth0` we are telling the CM4 to use that for XRCE-DDS somehow?

If not, please spell it out in laymans networking.

File: en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md (hunk 8 above)
Comment (hamishwillee) [reply en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:356]: So you're setting both mavlink and XRCE-DDS on the same local ethernet serial port?:

```
param set MAV_2_CONFIG     1000        # Configure Ethernet port
param set UXRCE_DDS_CFG    1000   
```

I don't think that is allowed - it shouldn't work as only one thing should own a serial port at a time.

Don't we normally have mavlink over a separate dedicated internal serial link?

Comment (farhangnaderi) [reply en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:356]: You are absolutely right Hamish! My bad in tests.

File: en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (hamishwillee) [reply en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md:234]: OK, so I'm just assuming I'm right - you could use Wifi for the comms,  but that would clash with internet. So we set up the local route.

**Comment by hamishwillee** [comment]
Thanks @farhangnaderi! All done. Merged :-)

**Comment by farhangnaderi** [comment]
> Thanks @farhangnaderi! All done. Merged :-)

Thank you!
